### PR TITLE
Homework6: Olga Chubryk

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/hometask6/The_Defenders.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
+"""It's the solution for task #1. This is the game "The_Warriors"
```
Comment (belovmd) [inline src/hometask6/The_Defenders.py:1]: 👍

File: src/hometask6/the_decorator_new.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+class TooManyErrors(Exception):
+    def __init__(self, function_name):
+        self.function_name = function_name
+
+    def __str__(self):
+        return 'TooManyErrors at {}'.format(self.function_name)
+
+
+def error_decorator(critical_error_count):
```
Comment (belovmd) [inline src/hometask6/the_decorator_new.py:19]: I expected it would  for one function call try to call it not more than critical_err_count times.

**Review by belovmd — COMMENTED** [review]
please address next issue

File: src/hometask6/the_decorator_new.py (hunk 2 above)
Comment (ghost) [reply src/hometask6/the_decorator_new.py:19]: Fixed

File: src/hometask6/the_decorator_new.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                           ''.format(name, result,
+                                     function.__invocation_count__)
+                except Exception:
+                    function.__errors_count__ += 1
+                    return 'Name: {},' \
+                           ' count function errors: {}' \
+                           ''.format(name, function.__errors_count__)
+            else:
+                try:
+                    if function.__errors_count__ > critical_error_count:
```
Comment (belovmd) [inline src/hometask6/the_decorator_new.py:43]: i don't see other options for it to go to while else condition

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**